### ISSUE TITLE
Shared dashboards refreshing

### ISSUE THREAD

**Issue by tiina303** [PostHog/posthog#10563]
## Is your feature request related to a problem?

User request:
We use PostHog to create usage dashboard shared with our customers

It looks like "Auto refresh" doesn't work if we close our dashboard page Is there a way to auto refresh the dashboard in the background? So that when our customer (that doesn't have an account in our PostHog space) opens the dashboard it's up to date

## Describe the solution you'd like

Two options:
1. Allow refreshes to be triggered from shared dashboards
2. Allow setting background refresh on shared dashboards

**Comment by jroberati** [comment]
It would be nice if shared dashboards triggered an update on page load if the refresh interval has passed.

**Comment by raghavmidha** [comment]
we have the same request. this would be a very helpful update

**Comment by wintrykiddo** [comment]
hi team, we just found posthog and it's AWESOME! it replaces the need to build dashboards with bootstrap and the like 🥹 and of course so much more. we're impressed and grateful.

however, our new posthog dashboards don't update automatically, which limits the degree of excitement they can create. hourly charts are over half a day old, which invalidates their use case.

i hereby submit my official request in addition to the other requesters to make them live updating regardless of any admin action 🫡

**Comment by mebeingken** [comment]
Agreed, it seems the analytics on dashboards should be automatically updated when shared via public url. We publish analytics to a multi-tenant app, and have dozens of dashboards to manage.

Is there any work around aside from manually touching each dashboard?

**Comment by camerondeleone** [comment]
+1 for a [Potential customer](https://app.hubspot.com/contacts/6958578/record/0-2/15581657270) requesting this.

**Comment by green-mint** [comment]
+1, we share dashboards with our customers, right now we are manually refreshing them but it would be great to have an auto refresh in the background every few hrs or so.

**Comment by mighty-peach** [comment]
+1, refreshing of shared dashboards would be super useful 🚀

**Comment by ordehi** [comment]
+1 https://posthoghelp.zendesk.com/agent/tickets/30843 (moved to PostHog: https://us.posthog.com/project/2/support/tickets/32422)

**Comment by dan-dr** [comment]
+1000

**Comment by DaveKeehl** [comment]
+1000000 😆

**Comment by nima-karimi** [comment]
closing as we have cache warming in place for shared insights viewed in the last 3 days, which will automatically refresh them when they get stale.

related PRs:
- https://github.com/postHog/posthog/pull/24388
- https://github.com/postHog/posthog/pull/27495

**Comment by andyzzhao** [comment]
To summarize there seems to be two pieces of feedback based on the ticket: https://posthoghelp.zendesk.com/agent/tickets/46636 (moved to PostHog: https://us.posthog.com/project/2/support/tickets/50184)

1. The user expects that setting a refresh interval on their interval view of the dashboard will also be the refresh interval on their shared dashboard. 
    - This sounds like a feature request to me but also makes me think the sharing concept is a bit confusing. 
2. Their shared dashboard is 18hr stale
    - this shouldn't happen if it was viewed with in the past three days

Currently, if a shared dashboard has been viewed in the past three days, we refresh it every hour.

However, I noticed UX pain with this which might be the reason for 2. If a dashboard has not been viewed in the past three days, the first view will be either stale data or no data. We kick off an async calculation in this case, but the client will need to refresh the page to see results.
